google_guest_agent/main.go

```diff
@@ -170,12 +171,48 @@ func run(ctx context.Context) {
 		},
 	}
 
+	// Only Enable sshtrustedca Watcher if osLogin is enabled.
+	// TODO: ideally we should have a feature flag specifically for this.
+	osLoginEnabled, _, _ := getOSLoginEnabled(newMetadata)
+	if osLoginEnabled {
+		eventsConfig.Watchers = append(eventsConfig.Watchers, sshtrustedca.WatcherID)
+	}
+
 	eventManager, err := events.New(eventsConfig)
 	if err != nil {
 		logger.Errorf("Error initializing event manager: %v", err)
 		return
 	}
 
+	eventManager.Subscribe(sshtrustedca.ReadEvent, nil, func(evType string, data interface{}, evData *events.EventData) bool {
+		// There was some error on the pipe watcher, just ignore it.
+		if evData.Error != nil {
+			return true
```

Comment:
Do we need debug log here?

Reply:
Yep, Debug makes sense. Will do. Thanks.

Reply:
done.